Backend/Hub/GroupHub.cs

```diff
@@ -18,40 +18,112 @@ public async Task<bool> Kick(string userId)
             UserSession session = await Database.UserSessions.FromSignalRId(Context.UserIdentifier);
             if (session == null) return false;
             
-            User query = await Database.Users.AsAsyncEnumerable().SingleAsync(b => b.Id == long.Parse(userId));
-            if (query.GroupMember.Group == null) return false; //Is default null?
+            User user = await Database.Users.AsAsyncEnumerable().SingleAsync(b => b.Id == long.Parse(userId));
+            if (user.GroupMember.Group == null) return false; //Is default null?
 
-            GroupMember gmem = query.GroupMember;
-            Group grp = query.GroupMember.Group;
+            GroupMember gmem = user.GroupMember;
+            Group grp = user.GroupMember.Group;
             grp.GroupMembers.Remove(gmem);
             Database.GroupMembers.Remove(gmem);
             grp.SyncPlacesWithUsers();
             await Database.SaveChangesAsync();
 
-            await Clients.Group(query.GroupMember.Group.SignalRId).UpdateGroupData(grp.GetFriendlyId(), await GetGroupMembers());
+            await Clients.Group(user.GroupMember.Group.SignalRId).UpdateGroupData(grp.GetFriendlyId(), await GetGroupMembers());
             
 
             //edge case if kicked get points and update challeng
-            if (query.GroupMember.IsDone)
+            if (user.GroupMember.IsDone)
             {
-                query.Score += grp.Challenge.Points;
+                user.Score += grp.Challenge.Points;
             }
             //edge case if kicked and last one then move group on
             if (grp.GroupMembers.All(b => b.IsDone))
             {
                 await grp.GetNewChallenge(Database.Challenges);
             }
             
+            await user.NewGroup(Database.Challenges, grp.Challenge.LongLat.Coordinate.X, grp.Challenge.LongLat.Coordinate.Y);
```

Comment:
If the order in NewGroup is latitude, longitude, then you need to reverse the X and Y since NetTopologySuite's order is long, lat as stated in the field's name. If NewGroup's order is long lat then you don't need to change anything.